src/seamless-immutable.js

```diff
@@ -471,8 +471,17 @@
     return makeImmutable(obj, mutatingObjectMethods);
   }
 
+  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
+  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
+
+  function isReactElement(obj) {
+    return obj.hasOwnProperty &&
+           obj.hasOwnProperty('$$typeof') &&
+           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
+  }
```

Comment:
What do you think about following [`React.isValidElement` implementation][1] more closely?

```js
function isReactElement (obj) {
  return (
    typeof obj === 'object' &&
    obj !== null &&
    obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE
  );
}
```

[1]: https://github.com/facebook/react/blob/932334d3d46627adb08013e8e/src/isomorphic/classic/element/ReactElement.js#L342-L347